test/functional/mining_validation_tests.py

```diff
@@ -0,0 +1,219 @@
+                             "5f0421987d5f0870000000000000000d2f6e6f64655374726174756d2f")
+
+def submit_scrypt2_maintance_blocks(node):
+    create_and_submit_block(node, 536870912, "005c4c63af85f3f991afc707e32dc6fbe7b9242f79fa1757df2bb1f37ceb18c1",
+                             "0a993f04e79234f2f130d139a53cadbd5e71240a626dd699ea03b2c4c449f35a", 1602066632, 4177526655,
+                             "6004c8987d5f0840000000000000000d2f6e6f64655374726174756d2f")
+
+    create_and_submit_block(node, 536870912, "1e5194b838e00dea864cb30a96162fbdd51cc2e0779a8a2c8374cbbb2c6c7a0c",
+                             "4b6338cc64ea6533eaf31daa146fc55e714ca5a813e3ef3c117580e325a56abc", 1602161828, 2863311370,
+                             "011104a40c7f5f0818000000000000000d2f6e6f64655374726174756d2f")
+
+    create_and_submit_block(node, 536870912, "a73c5e5c892ab1635730b0511f0db42adda29aaa40a5551e1bfdac012750520f",
+                             "c7bef233e5ae4dbd478ed2483a50a6b4e0eb174ed45ece8624a11db2f0fd3712", 1602161888, 2695539370,
+                             "011204e00c7f5f0818000000000000000d2f6e6f64655374726174756d2f")
+
+    create_and_submit_block(node, 536870912, "8a9f9c399f1de9d22179fb23ff865e1e01d7d532f20ca066a6ebbe3b099a501e",
+                             "f2b948814bc56ee53aa5a800900e92e4a53b517e7af27f5275df7a375cbc5043", 1602161901, 1213552085,
+                             "011304ed0c7f5f0818000000000000000d2f6e6f64655374726174756d2f")
+
+def submit_scrypt2_blocks(node):
+    create_and_submit_block(node, 1073741824, "ae25bbab4bafac5cb9c3c72d2cd933ea442f5690ddb0e8a825816c22da5f8373",
+                            "e1912560c8b66850c8ecacdfd3d672c8c3005f60c806d5640a470f1ad959f2d6", 1602161910, 1280660885,
+                            "011404fa0c7f5f0818000000000000000d2f6e6f64655374726174756d2f")
+
+    create_and_submit_block(node, 1073741824, "73a600249f7a4ef77f58703a7cc0d00d579c3240c0ff1c42a8f1ecbf5f59a1cf",
+                             "1b43cdc216316a3bd90f420f9a0eb723dea57fdb82e18a185dcd40b49423b400", 1602161933, 1414878485,
+                             "0115040d0d7f5f0818000000000000000d2f6e6f64655374726174756d2f")
+
+    create_and_submit_block(node, 1073741824, "122fb0df9260f5d8c585221cf9da6658a82fa72205423d35586cafeb14500891",
+                             "6b8b2572195e0c04b0605350f16936548f00d0dce8ee786cd0977f83fa988889", 1602161970, 1247106485,
+                             "011604420d7f5f0818000000000000000d2f6e6f64655374726174756d2f")
+
+    create_and_submit_block(node, 1073741824, "9a4138f568c294d6ceb4fad2a69a472f3baebaf8f2ee9facb808dfd4d4f238cb",
+                             "615153bc758cee5d5fb02bff559b366d91cbc0c38662514c0539bcae1845c725", 1602162036, 2829756970,
+                             "011704850d7f5f0818000000000000000d2f6e6f64655374726174756d2f")
+
+
+class MiningValidationTest(BitcoinSubsidiumTestFramework):
```

Comment:
there are no tests to check if a block with the wrong hash won't be accepted for each of network periods